h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -357,6 +384,55 @@ private void scrubMetaMap() {
         }
     }
 
+    public void addRestorePoint(long databaseVersion, RestorePoint rp) {
+        checkNotClosed();
+        restorePoints.setWriteVersion(currentVersion);
+        restorePoints.put(databaseVersion, rp);
+    }
+
+    public void removeRestorePoint(long databaseVersion) {
+        checkNotClosed();
+        restorePoints.setWriteVersion(currentVersion);
+        restorePoints.remove(databaseVersion);
+    }
+
+    long[] getRestorePointVersions() {
+        return restorePoints
+                .keySet()
+                .stream()
+                .mapToLong(Long::longValue)
+                .sorted()
+                .toArray();
+    }
+
+    public long getOldestRestorePointVersion() {
+        return restorePoints
+                .values()
+                .stream()
+                .mapToLong(it -> it.getOldestDatabaseVersionToKeep().getLong())
+                .findFirst()
+                .orElse(Long.MAX_VALUE);
```

Comment:
It should be `restorePoints.firstKey()`, isn't it? Why a separate property with oldest version is needed?